air/src/air/transition/frame.rs

```diff
@@ -3,80 +3,81 @@
 // This source code is licensed under the MIT license found in the
 // LICENSE file in the root directory of this source tree.
 
-use super::{FieldElement, Vec};
+use super::{super::Air, FieldElement, Vec};
+use utils::iterators::StreamingIterator;
 
-// EVALUATION FRAME
-// ================================================================================================
 /// A set of execution trace rows required for evaluation of transition constraints.
-///
-/// In the current implementation, an evaluation frame always contains two consecutive rows of the
-/// execution trace. It is passed in as one of the parameters into
+/// It is passed in as one of the parameters into
 /// [Air::evaluate_transition()](crate::Air::evaluate_transition) function.
+pub trait EvaluationFrame<E: FieldElement> {
+    /// Creates an empty frame
+    fn new<A: Air>(air: &A) -> Self;
+
+    /// Creates an frame instantiated from the provided rows
+    fn from_rows(rows: Vec<Vec<E>>) -> Self;
```

Comment:
We also could add a function `fn to_table(&self) -> Table<E>;` which would convert the frame into a table. This might come in handing when we deal with OOD composition.